### PR TITLE
when adding a sku set the skucode to the product Code + skucount if a

### PR DESCRIPTION
**Pull request by rmarchandTen24**
sku code isn't provided

### CODE REVIEW

File: model/entity/Product.cfc [hunk 1]
```diff
@@ -1018,6 +1018,13 @@ component displayname="Product" entityname="SlatwallProduct" table="SwProduct" p
 
 	// Skus (one-to-many)
 	public void function addSku(required any sku) {
+		//if sku code is null then create one automatically
+		if(isNull(arguments.sku.getSkuCode())){
+			var skusCount = this.getSkusCount();
+			var skuCode = this.getProductCode() & "-#skusCount + 1#";
+			arguments.sku.setSkuCode(skuCode);
+		}
+
```
Comment (sumitverma) [inline model/entity/Product.cfc]: I think we should do this when creating the sku and not in the bidirectional method. Here we are just assigning the sku to product.

Comment (rmarchandTen24) [inline model/entity/Product.cfc]: I disagree because when creating the sku you will not have index info on
where the item exists in the getSkus array

Comment (gregmoser) [inline model/entity/Product.cfc]: Also this doesn't work when you delete a sku.  Consider this:

ASDF-1
ASDF-2
ASDF-3
ASDF-4
ASDF-5

Then delete ASDF-3

ASDF-1
ASDF-2
ASDF-4
ASDF-5

Now add a new one...

Your logic would make it ASDF-5... but that already exists.

Comment (rmarchandTen24) [inline model/entity/Product.cfc]: https://github.com/ten24/slatwall/pull/3938

Comment (rmarchandTen24) [inline model/entity/Product.cfc]: I just thought of that so I think the logic only should run on new instances

**Pull request merged**